pkg/api/v1alpha1/cluster_webhook_test.go

```diff
@@ -1715,7 +1715,7 @@ func TestClusterValidateUpdateInvalidRequest(t *testing.T) {
 	cNew.Spec.ControlPlaneConfiguration.Count = cNew.Spec.ControlPlaneConfiguration.Count + 1
 	g := NewWithT(t)
 	err := cNew.ValidateUpdate(cOld)
-	g.Expect(err).To(MatchError(ContainSubstring("upgrading self managed clusters is not supported")))
+	g.Expect(err).To(MatchError(ContainSubstring("spec.ControlPlaneConfiguration: Forbidden: field is immutable")))
```

Comment:
So, if I'm understanding correctly, the worker node configs can be updated through the controller but the ControlPlaneConfiguration is still immutable, is that correct?

Reply:
Yes, exactly! Initially, the new full lifecycle controller completely ignored self managed clusters. But we recently added partial reconcile support by reacting to changes in the worker part of the spec in this [PR](https://github.com/aws/eks-anywhere/pull/3964)

Reply:
Alright, thanks for clarifying!